src/coreclr/jit/lsra.cpp

```diff
@@ -2756,8769 +2757,9059 @@ bool LinearScan::isMatchingConstant(RegRecord* physRegRecord, RefPosition* refPo
-                }
-            }
+            interval->physReg = assignedReg;
         }
-#ifdef FEATURE_SIMD
-#if FEATURE_PARTIAL_SIMD_CALLEE_SAVE
-        else if (currentInterval->isUpperVector)
+
+        // If there is no treeNode, this must be a RefTypeExpUse, in
+        // which case we did the reload already
+        if (treeNode != nullptr)
         {
-            // This is a save or restore of the upper half of a large vector lclVar.
-            Interval* lclVarInterval = currentInterval->relatedInterval;
-            assert(lclVarInterval->isLocalVar);
-            if (refType == RefTypeUpperVectorSave)
+            treeNode->gtFlags |= GTF_SPILLED;
+            if (treeNode->IsMultiReg())
             {
-                if ((lclVarInterval->physReg == REG_NA) ||
-                    (lclVarInterval->isPartiallySpilled && (currentInterval->physReg == REG_STK)))
-                {
-                    allocate = false;
-                }
-                else
-                {
-                    lclVarInterval->isPartiallySpilled = true;
-                }
+                treeNode->SetRegSpillFlagByIdx(GTF_SPILLED, currentRefPosition->getMultiRegIdx());
             }
-            else if (refType == RefTypeUpperVectorRestore)
+            if (spillAfter)
             {
-                assert(currentInterval->isUpperVector);
-                if (lclVarInterval->isPartiallySpilled)
+                if (currentRefPosition->RegOptional())
                 {
-                    lclVarInterval->isPartiallySpilled = false;
+                    // This is a use of lclVar that is flagged as reg-optional
+                    // by lower/codegen and marked for both reload and spillAfter.
+                    // In this case we can avoid unnecessary reload and spill
+                    // by setting reg on lclVar to REG_STK and reg on tree node
+                    // to REG_NA.  Codegen will generate the code by considering
+                    // it as a contained memory operand.
+                    //
+                    // Note that varDsc->GetRegNum() is already to REG_STK above.
+                    interval->physReg = REG_NA;
+                    writeLocalReg(treeNode->AsLclVar(), interval->varNum, REG_NA);
+                    treeNode->gtFlags &= ~GTF_SPILLED;
+                    treeNode->SetContained();
+                    // We don't support RegOptional for multi-reg localvars.
+                    assert(!treeNode->IsMultiReg());
                 }
                 else
                 {
-                    allocate = false;
+                    treeNode->gtFlags |= GTF_SPILL;
+                    if (treeNode->IsMultiReg())
+                    {
+                        treeNode->SetRegSpillFlagByIdx(GTF_SPILL, currentRefPosition->getMultiRegIdx());
+                    }
                 }
             }
         }
-        else if (r
```

Comment:
TODO: fix or remove?

Reply:
I don't see the reference line of your comment. Can you copy/paste the text where you see the leftover TODO?

Reply:
I think all the ones I flagged had your alias, so just search for that.